src/Data/Unrestricted/Linear.hs

```diff
@@ -0,0 +1,175 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+module Data.Unrestricted.Linear
+  ( -- * Unrestricted
+    -- $ unrestricted
+    Unrestricted(..)
+  , unUnrestricted
+    -- * Typeclasses for non-linear actions
```

Comment:
I'm not sure “non-linear" is quite appropriate here. I'd rather use something to the effect of

```suggestion
    -- * Duplicating and consuming value linearly
```